docs/determining-declared-license.md

```diff
@@ -0,0 +1,101 @@
+* If the registry information has a licenseUrl field, the ClearlyDefined summarizer extracts the license from that license url and sets the declared license to the extracted license
+* If the ClearlyDefined summarizer cannot extract the license from the license URL, it sets the declared license to NOASSERTION
+
+**Checking for the packageEntries field**
+
+* The ClearlyDefined summarizer then checks whether the registry information has a packageEntries field
+* If it does not, it leaves the declared license as it is
+* If it does have a packageEntries field, the ClearlyDefined summarizer creates a new definition with the files field set to those packageEntries
+
+**Merging definitions**
+
+* It then merges the definitions and, in the process, merges the declared licenses
+* If the original definition (prior to the merge) has a declared license of 'OTHER', it sets the declared license (on the merged definition) to the license on the new definition 
+* Otherwise, it combines the original definition license and the new definition license with AND
+
+### git
+* sources: https://github.com and https://gitlab.com
+* The crawler clones the repo for the component from either https://gitlab.com or https://github.com
+* TODO - how is the declared license decided?
+
+### crate
+* source: https://crates.io/
+* The crawler gets registry information from https://crates.io/api/v1/crates/
+* The ClearlyDefined summarizer sets the declared license to the license(s) in the registry information
+
+### deb
+* source: http://ftp.debian.org/
+* First, the crawler downloads a package file map from http://ftp.debian.org/debian/indices/package-file.map.bz2 and caches it (if there is not one already cached) for 8 hours [Code](https://github.com/clearlydefined/crawler/blob/f461b2358fbde130bcc5d183de01a4212c4cd66d/providers/fetch/debianFetch.js#L87)
+* It then pulls the registry information for the particular component from that package map file [Code](https://github.com/clearlydefined/crawler/blob/f461b2358fbde130bcc5d183de01a4212c4cd66d/providers/fetch/debianFetch.js#L114)
+* It then finds the relevant copyright URL from the registry information [Code](https://github.com/clearlydefined/crawler/blob/f461b2358fbde130bcc5d183de01a4212c4cd66d/providers/fetch/debianFetch.js#L295) [Example](https://metadata.ftp-master.debian.org/changelogs/main/0/0ad-data/0ad-data_0.0.17-1_copyright)
+* It then pulls information from the copyright URL [Code](https://github.com/clearlydefined/crawler/blob/f461b2358fbde130bcc5d183de01a4212c4cd66d/providers/fetch/debianFetch.js#L306)
+* And parses that information [Code](https://github.com/clearlydefined/crawler/blob/f461b2358fbde130bcc5d183de01a4212c4cd66d/providers/fetch/debianFetch.js#L320) to determine the declared license(s)
+* The ClearlyDefined summarizer then parses the declared licenses and, if necessary, joins them with 'AND'
```

Comment:
Does "if necessary" mean "if there is more than one"?

Reply:
That is correct - adding that in. Good catch!